dev/a11y_assessments/lib/use_cases/material_banner.dart

```diff
@@ -24,48 +24,62 @@ class MainWidget extends StatefulWidget {
   @override
   State<MainWidget> createState() => MainWidgetState();
 }
-
 class MainWidgetState extends State<MainWidget> {
-  double currentSliderValue = 20;
-  ScaffoldFeatureController<MaterialBanner, MaterialBannerClosedReason>? controller;
+
+  FocusNode dismissButtonFocusNode = FocusNode();
+  FocusNode showButtonFocusNode = FocusNode();
+
+  @override
+  void dispose() {
+    dismissButtonFocusNode.dispose();
+    showButtonFocusNode.dispose();
+    super.dispose();
+  }
+
+  void hideBanner() {
+    ScaffoldMessenger.of(context).hideCurrentMaterialBanner();
+    showButtonFocusNode.requestFocus();
+  }
+
+  void showBanner() {
+    ScaffoldMessenger.of(context).showMaterialBanner(
+      MaterialBanner(
+        padding: const EdgeInsets.all(20),
+        content: const Text('Hello, I am a Material Banner'),
+        leading: const Icon(Icons.agriculture_outlined),
+        backgroundColor: Colors.blue,
+        actions: <Widget>[
+          TextButton(
+            focusNode: dismissButtonFocusNode,
+            onPressed: () {
+              setState(() {
```

Comment:
This call to `setState` is not needed, as no state is updated. You could also use a function tear-off here for onPressed.

```suggestion
TextButton(
  focusNode: dismissButtonFocusNode,
  onPressed: hideBanner,
  child: const Text('DISMISS'),
)
```